pkg/logql/log/parser.go

```diff
@@ -486,3 +500,77 @@ func (u *UnpackParser) unpack(it *jsoniter.Iterator, entry []byte, lbs *LabelsBu
 	}
 	return entry, nil
 }
+
+type SyslogParser struct {
+	lbs      *LabelsBuilder
+	ntParser syslog.Parser // nontransparent parser
+	ocParser syslog.Parser // octetcounting parser
+	reader   *bytes.Reader
+}
+
+func NewSyslogParser() *SyslogParser {
+	p := &SyslogParser{
+		reader: bytes.NewReader([]byte{}),
+	}
+	p.ntParser = nontransparent.NewParser(syslog.WithListener(p.handleMessage))
+	p.ocParser = octetcounting.NewParser(syslog.WithListener(p.handleMessage))
+	return p
+}
+
+func (l *SyslogParser) Process(line []byte, lbs *LabelsBuilder) ([]byte, bool) {
+	if lbs.ParserLabelHints().NoLabels() {
+		return line, true
+	}
+	l.lbs = lbs
+	l.reader.Reset(line)
+
+	if line[0] == '<' {
```

Comment:
small nit: switch case would be bit nicer IMHO. e.g: one case for `nontransparent parser`, one for `octetcounting` parser and finally default one.